website/docs: clarify upgrade process (#15330)
goauthentik/authentik: website/docs/install-config/upgrade.mdx

authentik does not support downgrading. Make sure to back up your database in case you need to revert an upgrade.
:::

### Preview the release notes

Be sure to carefully read the [Release Notes](../releases/) for the specific version to which you plan to upgrade. The release might have special requirements or actions or contain breaking changes.

### Database backup

> Make a backup of your PostgreSQL database before upgrading. You can dump your existing database to get a backup file. For more information about dumping and backing up your database, refer to [Upgrade PostgreSQL on Docker Compose](../troubleshooting/postgres/upgrade_docker.md) or [Upgrade PostgreSQL on Kubernetes](../troubleshooting/postgres/upgrade_kubernetes.md).

Reviewer: I'm in this type of mood today: What do you think 

==

Before upgrading, make a backup of your PostgreSQL database. You can create a backup by dumping your existing database. For detailed instructions, refer to the relevant guide for your deployment method:

- [Upgrade PostgreSQL on Docker Compose](../troubleshooting/postgres/upgrade_docker.md)
- [Upgrade PostgreSQL on Kubernetes](../troubleshooting/postgres/upgrade_kubernetes.md)

### Upgrade sequence

> Upgrades need to follow the sequence of major releases and utilise the most recent minor release; **do not skip** directly from an older major version to the most recent version.

Reviewer: Mabye 

====

Upgrades must follow the sequence of major versions, using the latest available patch version at each step. **Do not skip major versions**.

For example, if you're currently on `2025.2.1`, upgrade in the following order:

1. Upgrade to the latest `2025.2.x`.
2. Then to the latest `2025.4.x`.
3. Finally to the latest `2025.6.x`.

Always use the latest patch version (`.x`) within each `major.minor` version before proceeding to the next major versio

For example, if you are currently running `2025.2.1`, you should first upgrade to the latest `2025.2.x` release, then to the latest `2025.4.x` release, and finally to the latest `2025.6.x` release, in sequence.

Always use the latest available patch version (_x_ in this case being the latest patch release) for each major.minor release.

### Outposts

> The versions of the authentik server and any authentik outposts must be the same. Always [upgrade any outposts](../add-secure-apps/outposts/upgrading.md) at the same time that you upgrade your authentik instance.

Reviewer: ```suggestion
The authentik server and all authentik outposts must run the same version. Be sure to [upgrade all outposts](../add-secure-apps/outposts/upgrading.md) at the same time as the main authentik instance.
```

Reviewer: "main authentik instance". Not sure about wording here

Author: "Core" maybe?

Reviewer: Looks good
